spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -267,6 +282,19 @@ protected void doStart() throws Exception {
 			this.taskExecutor = new SimpleAsyncTaskExecutor(
 					(getBeanName() == null ? "container" : getBeanName()) + "-");
 		}
+		long idleEventInterval = getIdleEventInterval();
+		if (idleEventInterval > 0) {
+			if (this.taskScheduler != null) {
+				afterPropertiesSet();
```

Comment:
I think you mean here `if (this.taskScheduler == null)`.
Otherwise we end up with an `NPE` in the next `this.taskScheduler.scheduleAtFixedRate()`.